### ISSUE TITLE
Extend data download 

### ISSUE THREAD

**Issue by Martin-Molinero** [QuantConnect/lean-cli#388]
- Extend data download to support other data providers apart from QC, similar to the backtest/research `--data-provider` option
     - First use for this feature is polygon

**Comment by Priyal-Saxena** [comment]
Hey, I researched about this.
I found option --data-provider and data_provider in research.py and backtest.py, as well as in test_backtest.py file. (but not in test_research.py)
I was hoping to add similar snippets of code to download.py and test_download.py files. 
but on reading code of download.py i came to the understanding that the user is being asked to select the dataset and/or product they want, and then to pay for the dataset.

<img width="668" alt="image" src="https://github.com/QuantConnect/lean-cli/assets/76258252/c30a14c7-4ffb-4907-a861-f4e7ba19762d">

so my doubt was that if we add an option to download data from other providers, will it still have 
1) options to select from
2) need for payment from the user
